ui/components/app/currency-input/currency-input.js

```diff
@@ -127,7 +130,8 @@ export default function CurrencyInput({
     if (shouldUseFiat()) {
       // Display ETH
       currency = preferredCurrency || ETH;
-      numberOfDecimals = 8;
+      numberOfDecimals =
+        primaryNumberOfDecimals <= 16 ? primaryNumberOfDecimals : 16;
     } else {
```

Comment:
@NiranjanaBinoy : 
Can you plz confirm that the input now does not allow entering > 16 decimal digits ?

I am not sure but may be we should stop user at 15th decimal digit [example](https://github.com/MetaMask/metamask-extension/pull/12140/files#diff-45a7e71158bc214b6ff3d28ad60cc509f25e80efdfff6ad59f7734d0e6246d66R33), at 16th digit input becomes controlled.

Reply:
@NiranjanaBinoy : can you plz check point above, other than that PR looks good to me.